frontend/src/app/data/calculation/calculation.service.ts

```diff
@@ -164,18 +164,20 @@ export class CalculationService implements OnDestroy {
     });
   }
 
-  public deleteResult(id: number){
+  public deleteResults(ids: number[]){
     const that = this;
```

Comment:
Note that the code you're referring to here (the function body) is, strictly, outside the actual changeset of this PR.
But since it's me who've introduced it previously, I'm happy to accomodate you with an answer to this question.

In the immediate arrow function scope, you're correct that the reference to `this` will be intact. 
For more clarity and directess, let's use the equivalent contextual reference to `this` when accessing the delete method (on row 170). (I made an amendment to this effect).

The reason for using the pattern is that we're emitting to the `resultDeletedSubscription` that is declared as a property of the service class (`this`).
The NextObserver passed to the "inner" subscription triggering the `delete` observable won't preserve `this`, which we need for accessing the `EventEmitter`. This is the reason for relying on block scope here.